code/modules/mob/living/simple_animal/hostile/bees.dm

```diff
@@ -0,0 +1,284 @@
+
+	var/image/wings
+	if(!bee_icons["[icon_base]_wings"])
+		bee_icons["[icon_base]_wings"] = image(icon = 'icons/mob/bees.dmi', icon_state = "[icon_base]_wings")
+	wings = bee_icons["[icon_base]_wings"]
+	overlays += wings
+
+
+//We don't attack beekeepers/people dressed as bees//Todo: bee costume
+/mob/living/simple_animal/hostile/poison/bees/CanAttack(atom/the_target)
+	. = ..()
+	if(!.)
+		return 0
+	if(ishuman(the_target))
+		var/mob/living/carbon/human/H = the_target
+		return !H.bee_friendly()
+
+
+/mob/living/simple_animal/hostile/poison/bees/Found(atom/A)
+	if(istype(A, /obj/machinery/portable_atmospherics/hydroponics))
+		var/obj/machinery/portable_atmospherics/hydroponics/Hydro = A
+		if(Hydro.seed && !Hydro.dead && !Hydro.recent_bee_visit)
+			wanted_objects |= /obj/machinery/portable_atmospherics/hydroponics //so we only hunt them while they're alive/seeded/not visisted
+			return 1
+	if(ishuman(A))
+		var/mob/living/carbon/human/H = A
+		return !H.bee_friendly()
+	return 0
+
+
+/mob/living/simple_animal/hostile/poison/bees/AttackingTarget()
+ 	//Pollinate
+	if(istype(target, /obj/machinery/portable_atmospherics/hydroponics))
+		var/obj/machinery/portable_atmospherics/hydroponics/Hydro = target
+		pollinate(Hydro)
+	else if(target == beehome)
+		var/obj/structure/beebox/BB = target
+		loc = BB
+		target = null
+		wanted_objects -= /obj/structure/beebox //so we don't attack beeboxes when not going home
+	else
+		if(beegent && isliving(target))
+			var/mob/living/L = target
+			beegent.reaction_mob(L, TOUCH)
+			L.reagents.add_reagent(beegent.id, rand(1,5))
+		target.attack_animal(src)
+
+
+/mob/living/simple_animal/hostile/poison/bees/proc/assign_reagent(datum/reagent/R)
+	if(istype(R))
+		beegent = R
+		name = "[initial(name)] ([R.name])"
+		generate_bee_visuals()
+
+
+/mob/living/simple_animal/hostile/poison/bees/proc/pollinate(obj/machinery/portable_atmospherics/hydroponics/Hydro)
+	if(!istype(Hydro) || !Hydro.seed || Hydro.dead || Hydro.recent_bee_visit)
+		target = null
+		return
+
+	target = null //so we pick a new hydro tray next FindTarget(), instead of loving the same plant for eternity
+	wanted_objects -= /obj/machinery/portable_atmospherics/hydroponics //so we only hunt them while they're alive/seeded/not visisted
+	Hydro.recent_bee_visit = TRUE
+	spawn(BEE_TRAY_RECENT_VISIT)
+		if(Hydro)
+			Hydro.recent_bee_visit = FALSE
+
+	var/growth = health //Health also means how many bees are in the swarm, roughly.
+	//better healthier plants!
+	Hydro.health += round(growth*0.5)
+	if(prob(BEE_POLLINATE_PEST_CHANCE))
+		Hydro.pestlevel = max(0, --Hydro.pestlevel)
+	if(prob(BEE_POLLINATE_YIELD_CHANCE)) //Yield mod is HELLA powerful, but quite rare
+		var/seed_yield = Hydro.seed.get_trait(TRAIT_YIELD)
+		Hydro.seed.set_trait(TRAIT_YIELD, seed_yield + 1, 10, 0)
+	if(prob(BEE_POLLINATE_POTENTCY_CHANCE))
+		var/seed_potency = Hydro.seed.get_trait(TRAIT_POTENCY)
+		Hydro.seed.set_trait(TRAIT_POTENCY, seed_potency + 1, 200, 0)
```

Comment:
Making the fix now, although I'm going to avoid only changing the yield_mod of the tray, as that is incredibly lackluster for something that happens as this does.